addons/project/models/project_task.py

```diff
@@ -1077,6 +1077,22 @@ def write(self, vals):
         if "personal_stage_type_id" in vals and not vals['personal_stage_type_id']:
             del vals['personal_stage_type_id']
 
+        # sends an email to the 'Task Creation' subtype subscribers
+        # When project_id is changed
+        if vals.get('project_id'):
+            project = self.env['project.project'].browse(vals.get('project_id'))
+            notification_subtype_id = self.env.ref('project.mt_project_task_new').id
+            partner_ids = project.message_follower_ids.filtered(lambda follower: notification_subtype_id in follower.subtype_ids.ids).mapped('partner_id.id')
+            if partner_ids:
+                for task in self:
+                    project_link = task.project_id._get_html_link(title=task.project_id.display_name)
+                    task.message_notify(
+                        body=_('Task Transferred from Project %s', project_link),
+                        partner_ids=partner_ids,
+                        email_layout_xmlid='mail.mail_notification_layout',
+                        record_name=task.display_name,
+                    )
```

Comment:
We could maybe do the link for each project before the for loop to avoid doing it for the same projects multiple times.

```suggestion
                link_per_project_id = {p.id: p._get_html_link(title=p.display_name) for p in self.project_id}
                for task in self:
                    project_link = link_per_project_id[task.project_id.id]
                    task.message_notify(
                        body=_('Task Transferred from Project %s', project_link),
                        partner_ids=partner_ids,
                        email_layout_xmlid='mail.mail_notification_layout',
                        record_name=task.display_name,
                    )
```